### PR TITLE
doc: wifi: Update memory footprint tables for 2.6 release

### PR DESCRIPTION
**Pull request by richabp**
Updated Wi-Fi memory footprint tables for 2.6 release.

Created new "Memory requirements for Wi-Fi applications in SoftAP mode" page for SoftAP mode.

NCSDK-26300

### CODE REVIEW

**Review by krish2718 — APPROVED** [review]

**Review by kittydepa — APPROVED** [review]

File: doc/nrf/protocols/wifi/sap_mode/mem_requirements_sap.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.. _ug_wifi_mem_req_softap_mode:
+
+Memory requirements for Wi-Fi applications in SoftAP mode
+#########################################################
+
+Code and RAM memory footprint requirements differ depending on the selected platform and the application example.
```
Comment (ioannisg) [inline doc/nrf/protocols/wifi/sap_mode/mem_requirements_sap.rst:6]:
```suggestion
Code and RAM memory footprint requirements differ depending on the selected platform, networking application layer, and application example.
```

Comment (ioannisg) [reply doc/nrf/protocols/wifi/sap_mode/mem_requirements_sap.rst:6]: Essentially, these examples differ with respect to the selected application layer...

File: doc/nrf/protocols/wifi/sap_mode/index.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.. _ug_wifi_softap_mode:
+
+SoftAP mode
+###########
+
+The following subpages describe the memory requirements of the SoftAP sample and how an nRF70 Series device operates in :term:`Software-enabled Access Point (SoftAP or SAP)` mode.
```
Comment (ioannisg) [inline doc/nrf/protocols/wifi/sap_mode/index.rst:6]: I'd say let's make this text more generics, as we are likely going to add more examples with SoftAP mode.

Comment (ioannisg) [inline doc/nrf/protocols/wifi/sap_mode/index.rst:6]:
```suggestion
The following subpages describe how an nRF70 Series device operates in :term:`Software-enabled Access Point (SoftAP or SAP)` mode, along with  the memory requirements of Wi-Fi® applications that use the SoftAP mode.
```

**Review by ioannisg — CHANGES_REQUESTED** [review]
Thanks for the PR @richabp 
I left some editorial comments in the review.

In addition to that, I had a look at the numbers and I see several inconsistencies.
Above all, we need to revisit the list of samples for each platform and operation mode. Scan and SoftAP seem to be correct. STA mode, however should not include any samples for nRF91 Series of ICs. Samples should be listed for nRF52840 DK and nRF7002 DK

**Comment by richabp** [comment]
> Thanks for the PR @richabp I left some editorial comments in the review.
> 
> In addition to that, I had a look at the numbers and I see several inconsistencies. Above all, we need to revisit the list of samples for each platform and operation mode. Scan and SoftAP seem to be correct. STA mode, however should not include any samples for nRF91 Series of ICs. Samples should be listed for nRF52840 DK and nRF7002 DK

As discussed, the team will check and update about the inconsistencies in the numbers. Then, another PR could be open after this gets merged.

**Comment by ioannisg** [comment]
The sample selection looks good now (MQTT sample still missing though)

**Comment by richabp** [comment]
> The sample selection looks good now (MQTT sample still missing though)

Will address thin in a doc improvement PR once this is merged.

**Review by ioannisg — APPROVED** [review]
+1 to get this in, since the sample selection is correct.
Footprint numbers are still wrong, but I assume they can be fixed before v2.6.0

**Comment by richabp** [comment]
> +1 to get this in, since the sample selection is correct. Footprint numbers are still wrong, but I assume they can be fixed before v2.6.0

@krga2022 @visinor FYI.

**Pull request merged**